script/tool/lib/src/firebase_test_lab_command.dart

```diff
@@ -105,183 +101,165 @@ class FirebaseTestLabCommand extends PluginCommand {
       }
     }
     _firebaseProjectConfigured!.complete(null);
   }
 
   @override
-  Future<void> run() async {
-    final Stream<Directory> packagesWithTests = getPackages().where(
-        (Directory d) =>
-            isFlutterPackage(d) &&
-            d
-                .childDirectory('example')
-                .childDirectory('android')
-                .childDirectory('app')
-                .childDirectory('src')
-                .childDirectory('androidTest')
-                .existsSync());
+  Future<List<String>> runForPackage(Directory package) async {
+    if (!package
+        .childDirectory('example')
+        .childDirectory('android')
+        .childDirectory('app')
+        .childDirectory('src')
+        .childDirectory('androidTest')
+        .existsSync()) {
+      printSkip('No example with androidTest directory');
+      return PackageLoopingCommand.success;
+    }
 
-    final List<String> failingPackages = <String>[];
-    final List<String> missingFlutterBuild = <String>[];
-    int resultsCounter =
-        0; // We use a unique GCS bucket for each Firebase Test Lab run
-    await for (final Directory package in packagesWithTests) {
-      // See https://github.com/flutter/flutter/issues/38983
+    final List<String> errors = <String>[];
 
-      final Directory exampleDirectory = package.childDirectory('example');
-      final String packageName =
-          p.relative(package.path, from: packagesDir.path);
-      _print('\nRUNNING FIREBASE TEST LAB TESTS for $packageName');
+    final Directory exampleDirectory = package.childDirectory('example');
+    final Directory androidDirectory =
+        exampleDirectory.childDirectory('android');
 
-      final Directory androidDirectory =
-          exampleDirectory.childDirectory('android');
+    // Ensures that gradle wrapper exists
+    if (!await _ensureGradleWrapperExists(androidDirectory)) {
+      errors.add('Unable to build example apk');
+      return errors;
+    }
 
-      final String enableExperiment = getStringArg(kEnableExperiment);
-      final String encodedEnableExperiment =
-          Uri.encodeComponent('--enable-experiment=$enableExperiment');
+    await _configureFirebaseProject();
 
-      // Ensures that gradle wrapper exists
-      if (!androidDirectory.childFile(_gradleWrapper).existsSync()) {
-        final int exitCode = await processRunner.runAndStream(
-            'flutter',
-            <String>[
-              'build',
-              'apk',
-              if (enableExperiment.isNotEmpty)
-                '--enable-experiment=$enableExperiment',
-            ],
-            workingDir: androidDirectory);
+    if (!await _runGradle(androidDirectory, 'app:assembleAndroidTest')) {
+      errors.add('Unable to assemble androidTest');
+      return errors;
+    }
 
-        if (exitCode != 0) {
-          failingPackages.add(packageName);
-          continue;
-        }
+    int resultsCounter = 0; // Use a unique GCS bucket for each run.
```

Comment:
It isn't clear how this comment relates to the proceeding code.

Reply:
Added more explanation.